internal/worker/liveness.go

```diff
@@ -0,0 +1,69 @@
+package worker
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.hollow.sh/toolbox/events"
+	"go.hollow.sh/toolbox/events/registry"
+
+	"github.com/nats-io/nats.go"
+)
+
+var (
+	once           sync.Once
+	checkinCadence = 30 * time.Second
+)
+
+// This starts a go-routine to peridocally check in with the NATS kv
+func (w *Worker) startWorkerLivenessCheckin(ctx context.Context) {
+	natsJS, ok := w.stream.(*events.NatsJetstream)
+	if !ok {
+		w.logger.Error("Non-NATS stores are not supported for worker-liveness")
+		return
+	}
+
+	once.Do(func() {
+		if err := registry.InitializeActiveControllerRegistry(natsJS); err != nil {
+			w.logger.WithError(err).Error("unable to initialize active worker registry")
+			return
+		}
+
+		go w.checkinRoutine(ctx)
```

Comment:
Do we want a `w.syncWG.Add(1)`, and a related `defer w.syncWG.Done()` in `checkinRoutine` ?